machines/openstack/worker.machineset.yaml

```diff
@@ -0,0 +1,72 @@
+---
+apiVersion: cluster.k8s.io/v1alpha1
+kind: MachineSet
+metadata:
+  name: worker
+  namespace: {{ .TargetNamespace }}
+  labels:
+    sigs.k8s.io/cluster-api-cluster: {{.OpenStack.ClusterName}}
+    sigs.k8s.io/cluster-api-machine-role: worker
+    sigs.k8s.io/cluster-api-machine-type: worker
+spec:
+  replicas: {{.OpenStack.Replicas}}
+  selector:
+    matchLabels:
+      sigs.k8s.io/cluster-api-machineset: worker
+      sigs.k8s.io/cluster-api-cluster: {{.OpenStack.ClusterName}}
+  template:
+    metadata:
+      labels:
+        sigs.k8s.io/cluster-api-machineset: worker
+        sigs.k8s.io/cluster-api-cluster: {{.OpenStack.ClusterName}}
+        sigs.k8s.io/cluster-api-machine-role: worker
+        sigs.k8s.io/cluster-api-machine-type: worker
+    spec:
+      providerConfig:
+        value:
+          apiVersion: openstack.cluster.k8s.io/v1alpha1
+          kind: OpenStackMachineProviderConfig
+          {{- if .OpenStack.WithCreds }}
+          credentialsSecret:
+            name: openstack-credentials-secret
+          {{- end}}
+          ami:
+            id: {{.OpenStack.Image}}
+            filters:
+              - name: "name"
+                values:
+                - CoreOS-{{.OpenStack.ReleaseChannel}}-{{.OpenStack.ContainerLinuxVersion}}-*
```

Comment:
These filters are consistent with what we're doing now for AWS.  But that approach is stale (openshift/machine-api-operator#84).  I'd rather drop this section until we want to come back and address RHCOS/oscontainer auto-updates.  There's more background on future plans in openshift/installer#281.

Reply:
agree, let's only rely on explicit image id for now

Reply:
below this line this is all specific for the aws actuator config https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/cloud/aws/providerconfig/v1alpha1/types.go
This will need to go in hand with the openstack actuator is there a link for that? Is this to be used with upstream one https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/cloud/openstack/openstackproviderconfig/v1alpha1/types.go?

Reply:
yup, removed these lines. Using just the ID now.